specs/IFramePermissionRequested.md

```diff
@@ -0,0 +1,363 @@
+# Background
+The WebView2 team has been asked to provide support for handling permission
+requests that come from iframes. These permission requests occur when content
+within the iframe are requesting access to priveleged resources. The permission
+request types that we support are: Microphone, Camera, Geolocation,
+Notifications, Other Sensors, and Clipboard Read.
+
+We currently have a [PermissionRequested](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2permissionrequestedeventargs?view=webview2-1.0.1020.30)
+event on our `CoreWebView2` which is raised for any permission requests
+(either from webview or iframes). However, our customers do not have a way to
+determine whether the request has come from the webview or an iframe and
+handle these cases seperately. As such, we plan to expand our
+`CoreWebView2Frame` API to include the `PermissionRequested` event.
+
+A current limitation of our `CoreWebView2Frame` API is that it only supports top
+level iframes. Any nested iframes will not have a `CoreWebView2Frame`
+associated with them.
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+# Description
+We propose extending `CoreWebView2Frame` to include the `PermissionRequested`
+event. This event will be raised whenever an iframe requests permission to
+priveleged resources.
+
+Additionally, we propose extending `CoreWebView2PermissionRequestedEventArgs`
+to include a `Handled` property.
+
+To maintain backwards compatibility, by default we plan to raise
+`PermissionRequested` on both `CoreWebView2Frame` and `CoreWebView2`. The
+`CoreWebView2Frame` event handlers will be invoked first,
+before the `CoreWebView2` event handlers. If `Handled` is set true as part of
+the `CoreWebView2Frame` event handlers, then the `PermissionRequested` event
+will not be raised on the `CoreWebView2`, and its event handlers will not be
+invoked.
+
+In the case of a nested iframe requesting permission, we will raise the event
+off of the top level iframe. This is due to the previously mentioned limitation
```

Comment:
I wouldn't say this is due to this limitation. The PermissionRequested event is raised for the frame and its children in CoreWebView2 so can be reasonable to do the same on CoreWebView2Frame. In the future we may add CoreWebView2Frames for all frames and wouldn't change this behavior.